release-20.1: build: touch lib/*.so files

otan opened the pull request:
TeamCity can't resolve empty directories as artifacts, so add dummy
files when doing mkrelease.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52171)
<!-- Reviewable:end -->

otan committed:
release-20.1: build: touch lib/*.so files

TeamCity can't resolve empty directories as artifacts, so add dummy
files when doing mkrelease.

Release note: None

sumeerbhola reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52171)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @sumeerbhola)

___
*[build/builder/mkrelease.sh, line 107 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52171#-MDoZ2oPCBwUC7Wrizg9:-MDoZ2oQ2nCTQ_5qOIiK:b-e206k3) ([raw file](https://github.com/cockroachdb/cockroach/blob/3d919d2132b0412ad8a1c92e6d4bf5f17b6c0d09/build/builder/mkrelease.sh#L107)):*
> ```Shell
> # lib is populated in v20.2 or higher, but we make a temporary directory
> # in /lib such that TeamCity can pick up the artifacts.
> (set -x && mkdir -p lib && touch lib/libgeos.so && touch lib/libgeos_c.so && CGO_ENABLED=1 make BUILDTYPE=release "${args[@]}" "$@")
> ```

it this touch to force the rebuild of this target? I assume not since these are dynamically linked. Could you update the comment to explain?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52171)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[build/builder/mkrelease.sh, line 107 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52171#-MDoZ2oPCBwUC7Wrizg9:-MDoshNr9KCK8Vh_pCKP:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/3d919d2132b0412ad8a1c92e6d4bf5f17b6c0d09/build/builder/mkrelease.sh#L107)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

it this touch to force the rebuild of this target? I assume not since these are dynamically linked. Could you update the comment to explain?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52171#-:-MDpfH3T805qylhvpuUS:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)



<!-- Sent from Reviewable.io -->
